### PR TITLE
fix: send all pushConfig when creating subscription

### LINKED ISSUE #1915: Incorrect Handling of Subscription Push Configuration
## Description:

I encountered an issue while implementing a webhook feature using the Google Pub/Sub SDK for Node.js. Specifically, I was trying to create a push subscription with message unwrapping and without metadata, as outlined in the official documentation. However, despite following the provided instructions, the push subscription was not created with the expected configurations.

Upon further investigation into the source code of the SDK, I discovered an issue in the way push configurations are handled. In the method formatMetadata_ within the subscription.ts file, there is a segment of code that handles the push configuration as follows:
```typescript
formatted.pushConfig = {
  pushEndpoint: metadata.pushEndpoint,
};
delete formatted.pushEndpoint;
```
his code snippet only includes the pushEndpoint property in the pushConfig object and removes any other properties. Consequently, when creating a subscription with additional push configuration options, such as message unwrapping (noWrapper), these options are not retained.

To address this issue, I propose a modification to the code to ensure that existing push configuration properties are preserved when setting the pushEndpoint. Here's the updated code snippet:
```typescript
if (metadata.pushEndpoint) {
  formatted.pushConfig = {
    ...metadata.pushConfig,
    pushEndpoint: metadata.pushEndpoint,
  };
  delete formatted.pushEndpoint;
}

```
This modification ensures that any existing push configuration options are retained and only the pushEndpoint property is overwritten.

### Steps to Reproduce:

Use the Google Pub/Sub SDK for Node.js to create a push subscription.
Attempt to include additional push configuration options, such as message unwrapping (noWrapper), in the subscription creation request.

### Expected Behavior:

The push subscription should be created with the specified push configuration options, including message unwrapping (noWrapper), as per the documentation.

### Actual Behavior:

The push subscription is created without the specified push configuration options, and only the pushEndpoint property is retained.

### Proposed Solution:

Modify the handling of push configuration in the formatMetadata_ method to ensure that existing push configuration properties are preserved when setting the pushEndpoint, as described above.

Thank you for your attention to this matter. I believe addressing this issue will improve the usability and reliability of the Google Pub/Sub SDK for Node.js.

#### Environment details

  - OS: MacOS Ventura 13.1
  - Node.js version: v22.0.0
  - npm version: 10.5.1 
  - `@google-cloud/pubsub` version: ^4.1.0

### PR DESCRIPTION
**Pull request by useCallback**
Fixes #1915  🦕

### CODE REVIEW

**Review by feywind — APPROVED** [review]

**Comment by feywind** [comment]
I think as an external PR this might need a second review... can I bug you for it, @kamalaboulhosn ?

**Review by kamalaboulhosn — APPROVED** [review]

**Pull request merged**
